### DISCUSSION TITLE
improved resolution

### DISCUSSION THREAD

**Discussion by joelkraehemann** [libsndfile/libsamplerate#198] [Show and tell]
I took your coefficients and some of your resampling code:

https://github.com/gsequencer/gsequencer/blob/5.5.x/ags/audio/ags_samplerate_coeffs.h
https://github.com/gsequencer/gsequencer/blob/5.5.x/ags/audio/ags_resample_util.c

Further, libsamplerate introduced artifacts from one sample to another.
 In some cases unwanted.

**Comment by erikd** [comment]
> Further, libsamplerate introduced artifacts from one sample to another.
> In some cases unwanted.

All resampling introduces artifacts. The question is how big were those artifacts?

Furthermore, `libsamplerate` is designed to be used on an (effectively) infinite stream as the input. If you are resampling short chunks of audio (ie up to 10s of seconds) you will likely achieve less than stellar results.

**Reply by joelkraehemann** [reply]
I was using the very same secret rabbit struct for different audio sample data. I.e. to compute resampled MIDI locale all with the same secret rabbit struct.
Actually for each key you would need a dedicated struct or empty the sample cache.

**Reply by erikd** [reply]
> very same secret rabbit struct for different audio sample data

That is probably not going to achieve your purpose.

I am a bit confused. What actual issue do you have and is it a problem with `libsamplerate` itself or how you are using it?

**Reply by joelkraehemann** [reply]
first it was just a dependency and then I needed to improve integration into existing concepts of GSequencer.

**Reply by erikd** [reply]
Yes, outside straight sample rate conversions (on effectively infinite streams of samples) you will need to mess about a bit. All you need to do is abide by the license (gsequencer is GPL so you are fine).
